File: RomanToInteger/RomToInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        summ= 0
        for i in range(len(s)-1,-1,-1):
            num = roman[s[i]]
            if 3*num < summ: 
                summ = summ-num
            else: 
                summ = summ+num
        return summ
    

File: RomanToInteger/RomToInt.js
var romanToInt = function (s) {
  const roman = {
    I: 1,
    V: 5,
    X: 10,
    L: 50,
    C: 100,
    D: 500,
    M: 1000,
  };
  let result = 0;
  for (let i = 0; i < s.length; i++) {
    if (roman[s[i]] < roman[s[i + 1]]) {
      result -= roman[s[i]];
    } else {
      result += roman[s[i]];
    }
  }
  return result;
};
